pkg/plugin/plugin.go

```diff
@@ -754,11 +755,67 @@ func (e *AutoscaleEnforcer) Score(
 	return score, nil
 }
 
-// ScoreExtensions is required for framework.ScorePlugin, and can return nil if it's not used
-func (e *AutoscaleEnforcer) ScoreExtensions() framework.ScoreExtensions {
+// NormalizeScore weights scores uniformly in the range [minScore, trueScore], where
+// minScore is framework.MinNodeScore + 1.
+func (e *AutoscaleEnforcer) NormalizeScore(
+	ctx context.Context,
+	state *framework.CycleState,
+	pod *corev1.Pod,
+	scores framework.NodeScoreList,
+) (status *framework.Status) {
+	ignored := e.state.conf.ignoredNamespace(pod.Namespace)
+
+	e.metrics.IncMethodCall("NormalizeScore", ignored)
+	defer func() {
+		e.metrics.IncFailIfNotSuccess("NormalizeScore", ignored, status)
+	}()
+
+	logger := e.logger.With(zap.String("method", "NormalizeScore"), util.PodNameFields(pod))
+	logger.Info("Handling NormalizeScore request")
+
+	if !e.state.conf.RandomizeScores {
+		logger.Info("randomizeScores not set, skipping randomizations")
+		return
+	}
```

Comment:
I think it might be easier / clearer / etc to implement this in `ScoreExtensions` — that way the metrics also show whether NormalizeScore is enabled. (and so we don't get a bunch of repeated log lines just to say "doing nothing because this feature isn't enabled")

Reply:
I see; that's better.